src/test/poll/test_poll.rs

```diff
@@ -0,0 +1,186 @@
+        if ready < 0 {
+            return Err("error: poll failed".to_string());
+        } else if ready > 0 {
+            return Err(format!(
+                "error: pipe marked readable. revents={}",
+                read_poll.revents
+            ));
+        }
+
+        /* Now put information in pipe to be read */
+        if fd_write(pfd_write).is_err() {
+            return Err("error: could not write to pipe".to_string());
+        }
+
+        /* Check again, should be something to read */
+        read_poll.fd = pfd_read;
+        read_poll.events = libc::POLLIN;
+        read_poll.revents = 0;
+        ready = unsafe { libc::poll(&mut read_poll as *mut libc::pollfd, 1, 100) };
+        if ready != 1 {
+            return Err(format!("error: poll returned {} instead of 1", ready));
+        }
+
+        if read_poll.revents & libc::POLLIN == 0 {
+            return Err(format!(
+                "error: read_poll has wrong revents: {}",
+                read_poll.revents
+            ));
+        }
+
+        /* Make sure we got what expected back */
+        if let Err(e) = fd_read_cmp(pfd_read) {
+            return Err(e);
+        }
+
+        /* success! */
+        return Ok(());
+    })
+}
+
+fn test_creat() -> Result<(), String> {
+    let test_file = b"testpoll.txt";
+    let test_file = std::ffi::CString::new(*test_file).unwrap();
+    let mut fd = unsafe {
+        libc::creat(
+            test_file.as_bytes_with_nul() as *const _ as *const libc::c_char,
+            0o644,
+        )
+    };
+
+    if fd < 0 {
+        return Err(format!("error: could not create {:?}", test_file));
+    }
+
+    test_utils::run_and_close_fds(&[fd], || {
+        /* poll will check when testpoll has info to read */
+        let mut read_poll = libc::pollfd {
+            fd: fd,
+            events: libc::POLLIN,
+            revents: 0,
+        };
+        let mut ready = unsafe { libc::poll(&mut read_poll as *mut libc::pollfd, 1, 100) };
+        if ready < 0 {
+            return Err("error: poll on empty file failed".to_string());
+        } else if ready == 0 {
+            /* Note: Even though the file is 0 bytes, has no data inside of it, it is still instantly
+             * available for 'reading' the EOF. */
+            return Err(format!(
+                "error: expected EOF to be readable from empty file. revents={}",
+                read_poll.revents
+            ));
+        }
+
+        let test_file = b"testpoll.txt";
+
+        /* write to file */
+        if fd_write(fd).is_err() {
+            return Err(format!("error: could not write to {:?}", test_file));
+        }
+
+        /* Check again, should be something to read */
+        fd = nix::fcntl::open(
+            test_file.as_ref(),
+            nix::fcntl::OFlag::O_RDONLY,
+            nix::sys::stat::Mode::empty(),
+        )
+        .map_err(|e| e.to_string())?;
+
+        read_poll.fd = fd;
+        read_poll.events = libc::POLLIN;
```

Comment:
You should close the old file descriptor before overwriting it, so this can change to:

```rust
test_utils::run_and_close_fds(&[fd], || {
    ...

    /* write to file */
    fd_write(fd)?;
}

/* Check again, should be something to read */
let fd = nix::fcntl::open(
    test_file.as_ref(),
    nix::fcntl::OFlag::O_RDONLY,
    nix::sys::stat::Mode::empty(),
)
.map_err(|e| e.to_string())?;

test_utils::run_and_close_fds(&[fd], || {
    read_poll.fd = fd;
    read_poll.events = libc::POLLIN;

    ...
}
```